tutorials/advanced-development/advanced-pieces-topics/adding-filters.md

```diff
@@ -1,40 +1,212 @@
+
+{% hint style="info" %}
+You can also use `q` or `search` as a query parameter to do a full-text search. *Tip:* often this is all users want.
+{% endhint %}
+
+Next we'll explore how to add a filter by tag. Later, we'll look at filtering by a join as well.
+
+Add this code to `lib/modules/people-pages/index.js`. Note that earlier you added this module to `app.js`, extending `apostrophe-pieces-pages`. Now you need to add some custom configuration:
 
 {% code-tabs %}
-{% code-tabs-item title="lib/modules/people/index.js" %}
+{% code-tabs-item title="lib/modules/people-pages/index.js" %}
 ```javascript
 module.exports = {
-  // Other configuration options, then...
-  addFilters: [
+  // We already set the "extend" option in app.js, or we'd need it here
+  // Specify the schema fields we want to be able to filter by
+  piecesFilters: [
     {
-      name: 'tags',
-      label: 'Tags'
+      name: 'tags'
     }
   ]
-};
+}
 ```
 {% endcode-tabs-item %}
 {% endcode-tabs %}
 
-You can also allow multiple tags to be selected, in which case pieces with at least one of those tags are displayed:
+Here you're asking `apostrophe-pieces-pages` to automatically populate `req.data.piecesFilters.tags` with an array of choices. And, you're also asking that `tags` be accepted via the query string in the URL (for example, `/people?tags=doctors`).
+
+Now you can take advantage of that:
 
 {% code-tabs %}
-{% code-tabs-item title="lib/modules/people/index.js" %}
+{% code-tabs-item title="lib/modules/people-pages/index.html" %}
+```markup
+{# Link to all the tags, adding a parameter to the query string #}
+<ul class="tag-filters">
+  {% for tag in data.piecesFilters.tags %}
+    <li><a href="{{ data.url | build({ tags: tag.value }) }}">{{ tag.label }}</a></li>
+  {% endfor %}
+</ul>
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+
+{% hint style="info" %}
+**"What's going on in this code?"** On a pieces index page, `data.url` always contains the current URL. We want to add a `tags` parameter to the query string. Apostrophe's `build` filter merges new query parameters with the URL. We can also remove a query parameter by passing the empty string as its value.
+
+Notice that there are separate `value` and `label` properties for each tag, even though they are the same. This pattern is used consistently for all fields we define filters for, including fields like joins or select fields where the value and the label can be quite different. This lets you write a single macro to handle many filters.
+{% endhint %}
+
+### Displaying counts for tags
+
+You can display counts for the choices, so users know how many items are available with a given tag. 
+
+1. Add the `counts: true;` property to the `piecesFilters` in `index.js`
```

Comment:
No semicolon

Reply:
Just for the record, I didn't put that semicolon there. ;)